var/spack/repos/builtin/packages/boxlib/package.py

```diff
@@ -36,12 +36,14 @@ class Boxlib(CMakePackage):
 
     depends_on('mpi')
 
+    variant('dims', default=3, description='Number of spatial dimensions')
```

Comment:
I think that is the recommended approach. If there are a set number of dimensions allowed, e.g. 1-3, then you should use `values=(1, 2, 3)`. If it can be any integer, see `netcdf` for how that is handled.